### PR TITLE
CPM-825: Display errors on structure on save and create

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**
Screenshot from 2022-12-13 10-53-02

Screenshot from 2022-12-13 10-52-30

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ const CreateOrEditGeneratorPage: React.FC<CreateOrEditGeneratorProps> = ({
 
   const [generatorCodeToDelete, setGeneratorCodeToDelete] = useState<IdentifierGeneratorCode | undefined>();
   const [isDeleteGeneratorModalOpen, openDeleteGeneratorModal, closeDeleteGeneratorModal] = useBooleanState();
+  const hasStructureErrors = validationErrors.filter(item => item?.path?.includes('structure'))?.length > 0;
```
Comment (Balamung) [inline components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx:51]: Delimiter is also defined on the structure tab, so I think we should also include errors where path = 'delimiter' in this test ?

**Review by antoinearrive — APPROVED** [review]

**Review by Balamung — APPROVED** [review]

**Review by Beninho — APPROVED** [review]

File: components/identifier-generator/front/src/feature/pages/CreateGeneratorPage.tsx [hunk 2]
```diff
@@ -16,37 +17,52 @@ const CreateGeneratorPage: React.FC<CreateGeneratorProps> = ({initialGenerator})
...
+      const validationErrors = validateIdentifierGenerator(generator);
+      if (validationErrors.length > 0) {
+        setValidationErrors(validationErrors);
+        return;
+      }
+      setValidationErrors([]);
+
+      mutate(generator, {
+        onError: error => {
+          // @ts-ignore
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/CreateGeneratorPage.tsx:43]: Why @ts-ignore ?

File: components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx [hunk 3]
```diff
@@ -77,16 +82,18 @@ const CreateOrEditGeneratorPage: React.FC<CreateOrEditGeneratorProps> = ({
 
   const onStructureChange = (structure: Structure) => {
     const updatedGenerator = {...generator, structure};
-    onChangeGenerator(updatedGenerator);
+    if (structure.length === 0) {
+      onChangeGenerator({...updatedGenerator, delimiter: null});
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx:86]: Why put delimiter back to null ?

**Review by pierallard — APPROVED** [review]

File: components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx (hunk 3 above)
Comment (cemorin) [reply components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx:86]: I remember doing it after a test with Pauline but I don't find any info on tickets or messages. So I'll double check with her

Comment (cemorin) [reply components/identifier-generator/front/src/feature/pages/CreateOrEditGeneratorPage.tsx:86]: I checked with her, and its a wanted behavior, when there is no properties, we reset the delimiter

**Pull request merged**